internal/authenticateflow/stateful.go

```diff
@@ -261,15 +262,44 @@ func (s *Stateful) RevokeSession(
 		return ""
 	}
 
+	// Note: session.Delete() cannot be used safely, because the identity
+	// manager expects to be able to read both session ID and user ID from
+	// deleted session records. Instead, we match the behavior used in the
+	// identity manager itself: fetch the existing databroker session record,
+	// explicitly set the DeletedAt timestamp, and Put() that record back.
+
+	res, err := s.dataBrokerClient.Get(ctx, &databroker.GetRequest{
```

Comment:
Can you use `Patch` instead?

Reply:
That's a good idea, but I think we can't, because the `deleted_at` field is part of the databroker `Record` message itself, rather than the data it contains (e.g. the `Session` message).

It might be worth a bigger conversation at some point about whether we should modify the `Patch` method to make it be able to delete a record.